### PR TITLE
MSBuild: don't use both Before and AfterTargets

### PR DESCRIPTION
**Pull request by rainersigwald**
Clarify the interaction between BeforeTargets and AfterTargets and don't encourage using them together, which rarely does what the author expected.

@ghogen I'm adding you to the internal mail thread on this for context.

### CODE REVIEW

**Comment by PRMerger-2** [comment]
@rainersigwald : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/msbuild/target-build-order.md [hunk 1]
```diff
@@ -99,12 +99,21 @@ Consider the following script.
...
 
+Alternately, specify the order as
+
+```xml
+<Target Name="Optimize" BeforeTargets="Link">
+    <Message Text="Optimizing" />
+</Target>
+```
+
+It is not generally useful to specify _both_ `BeforeTargets` and `AfterTargets` on the same target. As described in the next section, only the first target encountered will cause the new target to run.
```
Comment (ghogen) [inline docs/msbuild/target-build-order.md:115]:
```suggestion
It is not generally useful to specify *both* `BeforeTargets` and `AfterTargets` on the same target. As described in the next section, only the first target encountered will cause the new target to run.
```

**Comment by PRMerger-2** [comment]
@ghogen : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by ghogen** [comment]
#sign-off

**Review by ghogen — APPROVED** [review]

**Pull request merged**